lib/vanagon/platform.rb

```diff
@@ -117,6 +117,20 @@ def [](key)
       end
     end
 
+    # Get the output dir for packages. If the output_dir was defined already (by
+    # the platform config) then don't change it.
+    #
+    # @param target_repo [String] optional repo target for built packages defined
+    #   at the project level
+    # @return [String] relative path to where packages should be output to
+    def output_dir(target_repo = "")
+      if @output_dir
+        @output_dir
+      else
+        File.join(@os_name, @os_version, target_repo, @architecture)
+      end
```

Comment:
Also, I did validate that an empty string in `target_repo` will implicitly do the right thing when run through File.join:

```
>> target_repo = ""
=> ""
>> @os_name = 'debian'
=> "debian"
>> @os_version = 'jessie'
=> "jessie"
>> @architecture = 'amd64'
=> "amd64"
>> File.join(@os_name, @os_version, target_repo, @architecture)
=> "debian/jessie/amd64"
>>
```